ensenso_camera/src/virtual_object_handler.cpp

```diff
@@ -176,8 +176,9 @@ VirtualObjectHandler::VirtualObjectHandler(ensenso::ros::NodeHandle& nh, const s
   // Create publisher thread
   if (!markerTopic.empty())
   {
-    markerThread = std::thread([&]() {
-      VirtualObjectMarkerPublisher publisher{ nh,      markerTopic,  markerPublishRate,
+    auto nh_ptr = &nh;
+    markerThread = std::thread([=]() {
+      VirtualObjectMarkerPublisher publisher{ *nh_ptr, markerTopic,  markerPublishRate,
```

Comment:
You are right, the arguments have to be captured by value. 

Using a pointer, however, does not solve the problem as far as I understand. The pointer could also be invalidated by the time the thread starts.

What should work is making the `VirtualObjectMarkerPublisher` constructor take Nodehandle as value and move it into its member.

Suggestion:
```
  VirtualObjectMarkerPublisher(ensenso::ros::NodeHandle _nh, const std::string& topic, double publishRate,
                               const NxLibItem& objects, const std::string& frame, std::atomic_bool& _stop)
    : nh(std::move(_nh)), rate(publishRate), stop(_stop)
  {
```

Then we can use `nh` inside the lambda instead of using the pointer.